components/MechPage/FeeMetrics.jsx

```diff
@@ -0,0 +1,216 @@
+import { Chart } from 'react-google-charts';
+import useSWR from 'swr';
+
+const fetchMetrics = async () => {
+  try {
+    const result = await getFeeFlowMetrics();
+    if (!result) {
+      throw new Error('Failed to fetch metrics');
+    }
+    return {
+      collectedFees: result.collectedFees,
+      recievedFees: result.recievedFees,
+      unclaimedFees: result.unclaimedFees,
+      totalFees: result.totalFees,
+    };
+  } catch (error) {
+    console.error('Error in fetchMetrics:', error);
+    return null;
+  }
+};
+
+const usePersistentSWR = (key, fetcher) =>
+  useSWR(key, fetcher, {
+    revalidateIfStale: false,
+    revalidateOnFocus: false,
+  });
+
+export const FeeMetrics = () => {
+  const { data: metrics, error } = usePersistentSWR(
+    'FeeFlowMetrics',
+    fetchMetrics,
+  );
+
+  const formerData = useMemo(
+    () => [
+      {
+        id: 'total-fees',
+        label: 'Total Agent Fees Collected',
+        value: metrics?.totalFees || 0,
+        color: '#7a9cf7',
+      },
+      {
+        id: 'unclaimed',
+        label: 'Unclaimed Fees',
+        value: metrics?.unclaimedFees || 0,
+        color: '#90a1b9',
+      },
+      {
+        id: 'claimed',
+        label: 'Claimed Fees',
+        value: metrics?.collectedFees || 0,
+        color: '#5fb178',
+      },
+      {
+        id: 'received',
+        label: 'Fees Received',
+        value: metrics?.recievedFees || 0,
+        color: '#68bcce',
+      },
+      {
+        id: 'olas-burned',
+        label: 'OLAS Burned',
+        value: metrics?.olasBurned || 0,
+        color: '#dab2e4',
+      },
+    ],
+    [metrics],
+  );
+
+  const data = [
+    ['From', 'To', '', { role: 'tooltip', type: 'string', p: { html: true } }],
+    [
+      'Total Agent Fees Collected',
+      'Unclaimed Fees',
+      formerData[1].value,
+      `Total Agent Fees Collected → Unclaimed Fees | $${formerData[1].value} (${Number((formerData[1].value / formerData[0].value) * 100).toFixed(2)}%)`,
+    ],
+    [
+      'Unclaimed Fees',
+      '',
+      formerData[1].value,
+      `Total Agent Fees Collected → Unclaimed Fees | $${formerData[1].value} (${Number((formerData[1].value / formerData[0].value) * 100).toFixed(2)}%)`,
+    ],
+    [
+      'Total Agent Fees Collected',
+      'Claimed Fees',
+      formerData[2].value,
+      `Total Agent Fees Collected → Claimed Fees | $${formerData[2].value} (${Number((formerData[2].value / formerData[0].value) * 100).toFixed(2)}%)`,
+    ],
+    // Olas burned branch is represented as 10% for better visual presentation as 1% appears too small on screens
+    [
+      'Claimed Fees',
+      'OLAS Burned',
+      formerData[2].value * 0.1,
+      `Claimed Fees → OLAS Burned | $${formerData[2].value * 0.1} (1%)`,
+    ],
+    [
+      'Claimed Fees',
+      'Fees Received',
+      formerData[2].value * 0.9,
+      `Claimed Fees → Fees Received | $${formerData[2].value * 0.9} (99%)`,
+    ],
+  ];
```

Comment:
this is hard to review, would be nice to leave some comments on what it does/what it's for, or maybe move small pieces to functions for better readability